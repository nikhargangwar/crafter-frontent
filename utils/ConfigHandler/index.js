const ConfigHandler = (event,updateConfig,userConfig,setUserFeConfig)=>{
  if(!updateConfig || !setUserFeConfig) return;
  setUserFeConfig({
    ...userConfig,
    [updateConfig]:event.target.value
  });
};

export default ConfigHandler;