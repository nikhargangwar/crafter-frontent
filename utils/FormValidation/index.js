import validateEmail from './emailValidation';
import validatePassword from './passwordValidation';
import validateString from './stringValidation';
import validateInteger from './integerValidation';

export default { validateEmail, validatePassword , validateString, validateInteger};
