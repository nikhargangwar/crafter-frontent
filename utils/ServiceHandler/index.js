export  let REPOSITORY ={
  username:'',
  email:'',
  repositoryImageAddress : '',
  token:''
};
export const customReactFlow = {
  left:0,
  top:84,
};

let FRONTEND = {
  port:'',
  numberOfReplicas : '',
  name:''
};
let BACKEND = {
  port:'',
  numberOfReplicas:'',
  name:''
};
let DATABASE={
  port:'',
  numberOfReplicas:'',
  name:'',
  dbUser:'',
  dbPassword:'',
  schemaName :''
};

export function getSpaceSeparated(configurations){
  let spaceSeperatedConfig=  Object.keys(configurations).reduce((acc,configuration)=>{
    let spaceSeperated = configuration
      .replace(/([A-Z])/g,' $1')
      .split(' ')
      .map(word => word && word[0].toUpperCase() + word.slice(1))
      .join(' ');
    acc.push(spaceSeperated);
    return acc;
  },[]);
  return spaceSeperatedConfig;
}
export function camelize(str) {
  return str.replace(/(?:^\w|[A-Z]|\b\w)/g, function(word, index) {
    return index === 0 ? word.toLowerCase() : word.toUpperCase();
  }).replace(/\s+/g, '');
}

export function getConfiguration(service){
  switch(service.split(' ')[0]){
  case 'FRONTEND':
    return FRONTEND;
  case 'BACKEND':
    return BACKEND;
  case 'DATABASE':
    return DATABASE;
  }
}

export function updateConfiguration(service,configuration){
  if(!service){
    REPOSITORY=configuration;
  }
  switch(service.split(' ')[0]){
  case 'FRONTEND':
    FRONTEND=configuration;
    break;
  case 'BACKEND':
    BACKEND=configuration;
    break;
  case 'DATABASE':
    DATABASE=configuration;
  }
}

export function getCustomEvent(serviceName){
  return {
    dataTransfer:{
      getData:(key)=>{
        if(key === 'application/reactflow'){
          return 'microservice';
        }else if(key === 'nameOfNode'){
          return serviceName;
        }
      }
    },
    clientX: getRandomCoordinates(-300,600,3),
    clientY: getRandomCoordinates(-300,600,3)
  };
}


export function getRandomCoordinates(from, to, fixed) {
  return (Math.random() * (to - from) + from).toFixed(fixed) * 1;
}

export function convertKibanaToCapital(str) {
  let words = str.split('_');
  let capitalizedWords = words.map(word => word.charAt(0).toUpperCase() + word.slice(1));
  return capitalizedWords.join(' ');
}

export function convertCamelCaseToCapital(str) {
  let words = str.split(/(?=[A-Z])/);
  let capitalizedWords = words.map(word => word.charAt(0).toUpperCase() + word.slice(1));
  return capitalizedWords.join(' ');
}

export function separatePascalCase(str) {
  return str?.split(/(?=[A-Z])/).join(' ');
}

export function capitalizeEachWord(str) {
  return str?.split(' ').map(word => word.charAt(0).toUpperCase() + word.slice(1)).join(' ');
}
