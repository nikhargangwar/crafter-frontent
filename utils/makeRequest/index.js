import axios from 'axios';
// import { apiEndPoints } from '../../constants/';
import { ERROR_ROUTE } from '../../constants/routes';

const serviceMap={
  'FRONTEND':'FrontEnd',
  'BACKEND':'BackEnd',
  'DATABASE':'Database',
};

function getRequestBody(imageTokens){
  const services=[];
  const mapServiceName = {};
  const validServices = ['FRONTEND','BACKEND','DATABASE'];
  for (const key in localStorage) {
    const service_type = key.split(' ')[0];
    if(validServices?.indexOf(service_type) !== -1){
      const configuration = localStorage.getItem(key);
      let service={};
      service['service_type'] = serviceMap[service_type];
      service['configurations'] = JSON.parse(configuration).userConfig;
      service['customEnv']= JSON.parse(configuration).customConfig;
      service['connected_service']=[];
      console.log(service['customEnv']['offline_enabled']);
      if(!process.env.OFFLINE_ENABLED)
      {
        service['imageRepository'] = JSON.parse(configuration).dockerConfig;
        service['imageRepository']['token'] = imageTokens[key];
      }
      const serviceName = JSON.parse(configuration).userConfig.name;
      mapServiceName[key] = [serviceName,services.length];
      services.push(service);
    }
  }
  const connections =  JSON.parse(localStorage.getItem('connection'));
  const newConnections = {};
  if(connections){
    let index = 0;
    Object.keys(connections).map(service => {
      index = mapServiceName[service][1];
      newConnections[mapServiceName[service][0]]=[];
      connections[service].map(connected_service=>{
        newConnections[mapServiceName[service][0]].push(mapServiceName[connected_service][0]);
      });
      services[index]['connected_service'] = newConnections[mapServiceName[service][0]];
    });
  }
  return services;
}

const makeRequest = async (
  apiEndPoint,
  navigate,
  token,
  dynamicConfig = {}
) => {
  try {
    const requestDetails = {
      baseURL: apiEndPoint.baseURL,
      url: apiEndPoint.url,
      method: apiEndPoint.method,
      headers: {
        Authorization: token
      },
      ...dynamicConfig,
    };
    const {data} = await axios(requestDetails);
    return data;
  } catch (error) {
    if(navigate)
    {
      const errorStatus = error.response?.status;
      if (!errorStatus) {
        navigate(ERROR_ROUTE);
      } else {
        navigate(`${ERROR_ROUTE}/${errorStatus}`);
      }
      return null;
    }
    else{
      throw new Error(error);
    }
    
  }
};

const authMakeRequest = async (configObj) => {
  const { data } = await axios(configObj);
  return data;
};

export default { makeRequest, authMakeRequest, getRequestBody };
