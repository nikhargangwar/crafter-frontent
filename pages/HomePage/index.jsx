import React, { useEffect, useState } from 'react';
import { Button, InputAdornment, Tab, Tabs, TextField, Box, Tooltip, Divider } from '@mui/material';
import { Search, Add } from '@mui/icons-material';
import { makeRequest } from '../../utils';
import { apiEndPoints } from '../../constants';
import { useNavigate } from 'react-router-dom';
import { DeploymentStackModal, ClusterModal, WelcomeSidebar, Header, FileUpload} from '../../components';

function HomePage() {
  const [tabIndex, setTabIndex] = useState(0);
  const [showSidebar, setShowSidebar] = useState(false);
  const [blueprints, setBlueprints] = useState([]);
  const [selectedBlueprintId, setSelectedBlueprintId] = useState('');
  const [clusters, setClusters] = useState([]);
  const [selectedClusterId, setSelectedClusterId] = useState('');
  const [openModal, setOpenModal] = useState(false);
  const [query, setQuery] = useState('');
  const [openFileUploadModal, setOpenFileUploadModal] = useState(false);
  const navigate = useNavigate();
  const token = localStorage.getItem('jwt_token');

  useEffect(() => {
    makeRequest.makeRequest(apiEndPoints.GET_ALL_BLUEPRINTS(), navigate, token).then((response) => {
      setBlueprints(response);
    });
  }, [showSidebar]);

  useEffect (() => {
    makeRequest.makeRequest(apiEndPoints.GET_ALL_CLUSTERS(), navigate, token).then((response) => {
      setClusters(response);
    });
  }, [openModal, showSidebar]);

  const tabName = tabIndex === 0 ? 'Blueprints' : 'Clusters';

  return (
    <Box className='flex w-full h-screen'>
      {tabIndex === 0 && openModal && <DeploymentStackModal openModal={openModal} setOpenModal={setOpenModal} />} 
      {tabIndex === 1 && openModal && <ClusterModal openModal={openModal} setOpenModal={setOpenModal} />}
      <FileUpload openModal={openFileUploadModal} setOpenModal={setOpenFileUploadModal} setBluePrints={setBlueprints} />
      <Box className='h-screen w-full py-5 px-10'>
        <Box className='flex justify-between items-center'>
          <Header/>
        </Box>
        <Box className='w-full h-20 flex justify-center' >
          <Box className='flex gap-2 justify-between font-bold text-5xl'>
            <h1 style={{fontFamily:'Bebas Neue'}}> {'LET\'S'} </h1>
            <Box className='flex flex-col items-center'>
              <h1 style={{color:'#18CCCC', fontFamily:'Bebas Neue'}}>CREATE</h1>
              <Divider sx={{width:'50%',bgcolor:'#18CCCC',height:3}}/>
            </Box>
            
          </Box>
          
        </Box>
        <Box className='flex gap-8'>
          <Tabs value={tabIndex} sx={{fontSize:'30px'}} onChange={(event, newValue) => {setTabIndex(newValue); setShowSidebar(false);}}>
            <Tooltip title='Create a system design' placement='top'>
              <Tab label='Blueprints' sx={{fontFamily: 'Poppins', fontWeight: 800, fontSize: 20}}/>
            </Tooltip>
            <Tooltip title='Configure options for deployment to clusters' placement='top'>
              <Tab label='Clusters' sx={{fontFamily: 'Poppins', fontWeight: 800, fontSize: 20}}/>
            </Tooltip>
          </Tabs>
        </Box>
        <Box className='flex justify-between items-end mt-10'>
          <Box
            onClick={() => setOpenModal(true)}
            sx={{
              borderRadius:3,
              border:'4px solid',
              borderColor:'primary.boxBorder',
              boxShadow: '1px 1px 10px 0.5px #fff'
            }} 
            className = 'flex flex-col bg-transparent justify-center items-center w-48 h-40 cursor-pointer hover:-translate-y-2'
            data-testid='new-button'
          >
            <Add sx = {{fontSize: 72}}/>
            <h2>New {tabName.slice(0,-1)}</h2>
          </Box>
          {tabIndex === 0 && <Button 
            sx={{bgcolor:'primary.main',color:'white',fontWeight:'bold',width:'120px',height:'40px',fontSize:'18px'}} 
            variant='contained'
            onClick={() => { setOpenFileUploadModal(true); }}>
              Import
          </Button>}
        </Box>
        <Box className='flex justify-between mt-16'>
          <Box sx={{width:'40%', fontWeight:'bold'}}>
            <h1 className='text-2xl font-extrabold font-[Poppins] mb-3'>{!tabIndex ? 'RECENT BLUEPRINTS' : 'CONFIGURATIONS'}</h1>
            <hr style={{ backgroundColor:'#18CCCC', height: '3px', border: 'none' }} />
          </Box>
          <TextField sx={{border:'2px solid', borderRadius:'5px', fontSize:'20px',fontWeight:'bold'}} placeholder ='Search' variant="outlined" size='small'
            onChange={(e) => setQuery(e.target.value)}
            type='search'
            InputProps={{
              startAdornment: (
                <InputAdornment position="start">
                  <Search />
                </InputAdornment>
              )
            }} />
        </Box>
        <Box className= 'flex flex-wrap justify-start gap-8 mt-10'>
          {tabIndex === 0 && blueprints
            .filter((blueprint) => {
              if (query === '') return blueprint;
              else if (blueprint.blueprintName.toLowerCase().includes(query.toLowerCase())) return blueprint;
            })
            .map((blueprint) => (
              <Box 
                sx={{ boxShadow: '1px 1px 10px 0.5px #18CCCC',border:'3px solid', borderColor:'primary.main'}} 
                key={blueprint.id} 
                data-testid='recent-blueprints'
                onClick = {() => {
                  setShowSidebar(true); setSelectedBlueprintId(blueprint.id);
                }} 
                className = 'flex flex-col justify-center items-center w-48 h-40 border-primary border-2 rounded-2xl shadow-tile cursor-pointer hover:-translate-y-2'
              >
                <h1 className='text-lg font-bold'>{blueprint.blueprintName}</h1>
              </Box>
            ))}
          {tabIndex === 1 && clusters
            .filter((cluster) => {
              if (query === '') return cluster;
              else if (cluster.clusterName.toLowerCase().includes(query.toLowerCase())) return cluster;
            })
            .map((cluster) => (
              <Box
                sx={{ boxShadow: '1px 1px 10px 0.5px #18CCCC',border:'3px solid', borderColor:'primary.main'}}
                key={cluster.id}
                data-testid='recent-clusters'
                onClick = {() => {
                  setShowSidebar(true); setSelectedClusterId(cluster.id);
                }}
                className = 'flex flex-col justify-center items-center w-48 h-40 border-primary border-2 rounded-2xl shadow-tile cursor-pointer hover:-translate-y-2'
              >
                <h1 className='text-lg font-bold'>{cluster.clusterName}</h1>
              </Box>
            ))}
          {/* <Box sx={{ boxShadow: '1px 1px 10px 0.5px #18CCCC',border:'3px solid', borderColor:'primary.main'}} className = 'flex flex-col justify-center items-center w-48 h-40 border-2 rounded-2xl cursor-pointer hover:-translate-y-2'>
            <h1 className='text-lg font-bold'>...</h1>
            <h1 className='text-lg font-bold'>All Blueprints</h1>
          </Box> */}
        </Box>
      </Box>
      {tabIndex === 0 &&  showSidebar && <WelcomeSidebar setShowSidebar={setShowSidebar} selectedBlueprintId={selectedBlueprintId}/>}
      {tabIndex === 1 &&  showSidebar && <WelcomeSidebar setShowSidebar={setShowSidebar} selectedClusterId={selectedClusterId}/>}
    </Box>
  );
}

export default HomePage;
