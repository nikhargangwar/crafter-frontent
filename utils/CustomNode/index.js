/* eslint-disable */
import React, { memo } from 'react';
import './CustomNode.css';
import { Handle, Position } from 'reactflow';
import propTypes from 'prop-types';
import reactIcon from '../../assets/REACT-3d.png';
import nodeIcon from '../../assets/EXPRESS-3d.png';
import postgresIcon from '../../assets/postgres-3d.png';
import { useEffect } from 'react';
import { useState } from 'react';
import makeRequest from '../makeRequest';
import { apiEndPoints } from '../../constants';
import { useNavigate } from 'react-router-dom';

// eslint-disable-next-line react/display-name, react/prop-types
export default memo(({ data}) => {
  const service = data.label.split(' ')[0];
  const pluginId = data.pluginId;
  const navigate = useNavigate();

  const [imgSrc,setImgSrc] = useState();
useEffect(()=>{
  makeRequest.makeRequest(apiEndPoints.GET_PLUGIN_BY_ID(pluginId), navigate, localStorage.getItem('jwt_token'))
  .then((response)=>{
    // setBlueprintName( response.metadata.name);
    console.log('hjhj',response.metadata.icon_node_filename);
    setImgSrc(response.metadata.icon_node_filename);
  });

},[imgSrc])
  // useEffect(() => {
  //   switch (service) {
  //     case 'REACT-FRONTEND':
  //       setImgSrc(reactIcon);
  //       break;
  //     case 'EXPRESS-BACKEND':
  //       setImgSrc(nodeIcon);
  //       break;
  //     case 'POSTGRESQL-DATABASE':
  //       setImgSrc(postgresIcon);
  //       break;
  //     default:
  //       setImgSrc(postgresIcon);
  //       break;
  //   }
  // }, [imgSrc]);

  return (

  <div className='node'>
      <div className="node-handles">
            <Handle type="source" style = {{width:'10px',height:'10px' }} position={Position.Top} id="a" />
            <Handle type="source" style = {{width:'10px',height:'10px' }}position={Position.Right} id="b" />
            <div className="node-content">
    
            <Handle type="source" style = {{width:'10px',height:'10px' }}position={Position.Bottom} id="c" />
            <Handle type="source" style = {{width:'10px',height:'10px' }}position={Position.Left} id="d" />
            <img src={`${process.env.REACT_APP_API_GATEWAY_URL}/minio/images/${imgSrc}`} alt="microservice-icon" />
       
       </div>
      </div>
      <div className="node-name">
      {
            data.label
          }
      </div>
      
  </div>
   

  
  );
});

