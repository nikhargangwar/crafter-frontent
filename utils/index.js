import CustomNode from './CustomNode';
import FormValidation from './FormValidation';
import makeRequest from './makeRequest';
import {camelize, getConfiguration, getSpaceSeparated, updateConfiguration} from './ServiceHandler';
export { CustomNode, FormValidation, makeRequest, camelize, getConfiguration, getSpaceSeparated, updateConfiguration };