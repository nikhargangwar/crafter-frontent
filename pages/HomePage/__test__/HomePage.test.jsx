import { fireEvent, render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import HomePage from '..';
import { makeRequest } from '../../../utils';

beforeEach(() => {
  jest.spyOn(JSON, 'parse').mockReturnValue({fullName: 'test'});
});

describe('HomePage', () => {

  it('should render correctly', () => {
    const { asFragment } = render(<Router><HomePage /></Router>);
    expect(asFragment()).toMatchSnapshot();
  });

  describe('Blueprint tab', () => {

    it('should open modal when New Blueprint button is clicked', async () => {
      jest.spyOn(makeRequest, 'makeRequest').mockResolvedValue([]);
      render(<Router><HomePage /></Router>);
      const button = screen.getByTestId('new-button');
      expect(button).toBeTruthy();
      await fireEvent.click(button);
      const modalHeading = screen.findByRole('heading', { name: 'Deployment Stack' });
      expect(modalHeading).toBeTruthy();
    });

    it('should open sidebar when a blueprint is clicked', async () => {
      jest.spyOn(makeRequest, 'makeRequest')
        .mockResolvedValueOnce([
          {
            id:'1',
            blueprintName: 'test',
          }
        ])
        .mockResolvedValueOnce([
          {
            id:'1',
            blueprintName: 'test',
          }
        ])
        .mockResolvedValueOnce({
          metadata: {
            name: 'test',
            description: 'test',
            version: 'test',
          },
          versions: ['test']
        })
        .mockResolvedValue({});
      render(<Router><HomePage /></Router>);
      const blueprint = await screen.findByTestId('recent-blueprints');
      expect(blueprint).toBeTruthy();
      fireEvent.click(blueprint);
      const sidebarHeading = screen.findByRole('heading', { name: 'BLUEPRINT INFORMATION' });
      expect(sidebarHeading).toBeTruthy();  
    });

    it('should display all recent blueprints', async () => {
      jest.spyOn(makeRequest, 'makeRequest').mockResolvedValue([
        {
          id:1,
          blueprintName: 'test1',
        },
        {
          id:2,
          blueprintName: 'test2',
        },
        {
          id:3,
          blueprintName: 'test3',
        },
      ]);
      
      render(<Router><HomePage /></Router>);
      const rows = await screen.findAllByText(/test/);
      expect(rows).toHaveLength(3);
    });
  });

  describe('Cluster tab', () => {
    
    it('should open modal when New Cluster button is clicked', async () => {
      jest.spyOn(makeRequest, 'makeRequest').mockResolvedValue([]);
      render(<Router><HomePage /></Router>);
      const clusterTab = await screen.findByRole('tab', { selected: false });
      fireEvent.click(clusterTab);
      const button = screen.getByTestId('new-button');
      expect(button).toBeTruthy();
      await fireEvent.click(button);
      const modalHeading = screen.findByRole('heading', { name: 'New Cluster' });
      expect(modalHeading).toBeTruthy();
    });
    
    it('should open sidebar when a cluster is clicked', async () => {
      jest.spyOn(makeRequest, 'makeRequest')
        .mockResolvedValueOnce([
          {
            id:'1',
            clusterName: 'test',
          }
        ])
        .mockResolvedValueOnce([
          {
            id:'1',
            clusterName: 'test',
          }
        ])
        .mockResolvedValueOnce({
          metadata: {
            name: 'test',
            description: 'test',
          }
        })
        .mockResolvedValue({});
      render(<Router><HomePage /></Router>);
      const clusterTab = await screen.findByRole('tab', { selected: false });
      fireEvent.click(clusterTab);
      const cluster = await screen.findByTestId('recent-clusters');
      expect(cluster).toBeTruthy();
      fireEvent.click(cluster);
      const sidebarHeading = screen.findByRole('heading', { name: 'CLUSTER INFORMATION' });
      expect(sidebarHeading).toBeTruthy();
    });

    it('should display all clusters', async () => {
      jest.spyOn(makeRequest, 'makeRequest').mockResolvedValue([
        {
          id:1,
          clusterName: 'test1',
        },
        {
          id:2,
          clusterName: 'test2',
        },
        {
          id:3,
          clusterName: 'test3',
        },
      ]);
      render(<Router><HomePage /></Router>);
      const clusterTab = await screen.findByRole('tab', { selected: false });
      fireEvent.click(clusterTab);
      const rows = await screen.findAllByText(/test/);
      expect(rows).toHaveLength(3);
    });
  });
});

