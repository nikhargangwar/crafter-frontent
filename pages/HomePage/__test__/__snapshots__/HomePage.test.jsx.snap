// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should render correctly 1`] = `
<DocumentFragment>
  <div
    class="flex w-full h-screen MuiBox-root css-0"
  >
    <div
      class="h-screen w-full py-5 px-10 MuiBox-root css-0"
    >
      <div
        class="flex justify-between items-center MuiBox-root css-0"
      >
        <div
          class="flex items-center justify-between p-3 h-[10%] w-full MuiBox-root css-0"
        >
          <div
            class="flex items-center w-14 ml-14 cursor-pointer"
          >
            <img
              src="crafter-logo-dark.svg"
            />
          </div>
          <div
            class="user"
          >
            <div
              class="MuiBox-root css-j0tqbx"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-whjmaz-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="Brightness4Icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <button
              aria-haspopup="true"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-137ul40-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <div
                class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault avatar css-1d090hw-MuiAvatar-root"
              >
                t
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="w-full h-20 flex justify-center MuiBox-root css-0"
      >
        <div
          class="flex gap-2 justify-between font-bold text-5xl MuiBox-root css-0"
        >
          <h1
            style="font-family: Bebas Neue;"
          >
             LET'S 
          </h1>
          <div
            class="flex flex-col items-center MuiBox-root css-0"
          >
            <h1
              style="color: rgb(24, 204, 204); font-family: Bebas Neue;"
            >
              CREATE
            </h1>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-i7rfwh-MuiDivider-root"
            />
          </div>
        </div>
      </div>
      <div
        class="flex gap-8 MuiBox-root css-0"
      >
        <div
          class="MuiTabs-root css-h2jtf5-MuiTabs-root"
        >
          <div
            class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
            style="overflow: hidden; margin-bottom: 0px;"
          >
            <div
              class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
              role="tablist"
            >
              <button
                aria-label="Create a system design"
                aria-selected="true"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1a0tgcb-MuiButtonBase-root-MuiTab-root"
                data-mui-internal-clone-element="true"
                role="tab"
                tabindex="0"
                type="button"
              >
                Blueprints
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-label="Configure options for deployment to clusters"
                aria-selected="false"
                class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1a0tgcb-MuiButtonBase-root-MuiTab-root"
                data-mui-internal-clone-element="true"
                role="tab"
                tabindex="-1"
                type="button"
              >
                Clusters
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <span
              class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
              style="left: 0px; width: 0px;"
            />
          </div>
        </div>
      </div>
      <div
        class="flex justify-between items-end mt-10 MuiBox-root css-0"
      >
        <div
          class="flex flex-col bg-transparent justify-center items-center w-48 h-40 cursor-pointer hover:-translate-y-2 MuiBox-root css-ip4xcc"
          data-testid="new-button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-f29ca8-MuiSvgIcon-root"
            data-testid="AddIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
          <h2>
            New Blueprint
          </h2>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-igfdmi-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Import
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="flex justify-between mt-16 MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-1z0rujf"
        >
          <h1
            class="text-2xl font-extrabold font-[Poppins] mb-3"
          >
            RECENT BLUEPRINTS
          </h1>
          <hr
            style="background-color: rgb(24, 204, 204); height: 3px;"
          />
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root css-1hwz93b-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-ittuaa-MuiInputAdornment-root"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
            </div>
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputTypeSearch MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedStart css-1xp6qmi-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r2:"
              placeholder="Search"
              type="search"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="flex flex-wrap justify-start gap-8 mt-10 MuiBox-root css-0"
      />
    </div>
  </div>
</DocumentFragment>
`;
